roles/bootstrap/tasks/main.yml

```diff
@@ -1,16 +1,26 @@
-#
+
 # the order of tasks
 # run just one using the tag:
 # ansible-playbook -i inventory.yml bootstrap.yml --vault-password-file ~/.vault_pass.txt --tags "stage7"
 
 
 # copy and extract are no longer needed and will be removed soon
+#####- include: bundle_copy.yml     #<-- no used as necro is one big tarball now
+###### - include: bundle_extract.yml #<-- no used as necro is one big tarball now
 
+#############################################################
 - include: set_date_manually.yml
 - include: add_ansible_user.yml
-- include: fetch_bundles.yml 
-##- include: copy_bundles.yml   
-##- include: extract_bundles.yml
-- include: install_bundles.yml
+- include: bundle_fetch.yml 
+- include: bundle_prepare.yml
+- include: bundle_install.yml 
+
+#  When bundle_install python fails due to timeout- 
```

Comment:
If we expect this timeout, let's have it drop a stone just before installing python, print this message if the stone's found, and delete the stone as the next step.

Reply:
How sure are you it is really a timeout happening here? (Wasn't timeout already very high?) Does Ansible report timeout, or does it hang?

I'm experiencing the latter with Wget and Python installs to an Indy. Weirdly OpenSSH installs OK.

My Irix and Ansible skills are not too good, but it looks like inst somehow hangs. Ansible does not report a timeout. Running the install script manually works, OR if I try observing inst output by redirecting it (in the nekoware install script) to a log file.

Reply:
It's the python timeout waiting for something back from telnet. It's not ansible. To get around this you need to custom compile a local python that has a timeout greater than 300 seconds (5min).

Reply:
This can all be fixed with RSE, so I'm not so inclined to make it more fixed than a comment like this in the code. INST is just way too slow to deal with local python timeouts waiting for some output from it.

Reply:
According to Ansible documentation, the timeout in ansible.cfg timeout is SSH connection timeout.

The timeout in task telnet commands is for telnet connection. However, to me it looks like this is for **connection attempt** only. From telnet plugin sources one can see it is **not** used in expect when waiting for the prompt after a command. Expect timeout as added to Ansible later.

I am starting to suspect this problem is not exactly a timeout but something bit different.

Reply:
Thanks for explanation, and sorry for my (possibly confusing) follow-up comment which I posted before updating/noticing it.

If the timeout happens at Python socket level the timeout provided to telnetlib is somehow ignored (?) and inst really doesn't give output for such long time, pretty much the only thing to do would be make Ansible telnet plugin to timeout before it hangs (so at least there would be an error) and install dependencies in smaller batches. Or install only OpenSSH using telnet, and use it to install other packages.

But I agree, those would be bit fruitless changes if this is soon replaced anyway.

Reply:
Maybe the simplest "fix" here would be redirect install script command outputs (e.g. "/usr/nekoware/dist/install_python.sh > /tmp/install_python.log").

I do not really understand what exactly is happening. Ansible telnet even got the following # prompt, which is the most confusing part here as obviously the telnet connection is still open.

If this sounds OK to you I could still verify it even more properly in evening (with Indy and 6.5.22) and add it to the pull request.

Reply:
This should be investigated on its own smaller branch and PR.